doc/design.md

```diff
@@ -0,0 +1,136 @@
+# agentmon Unmasked
+
+Fundamentally, `agentmon` is a silly little server that proxies metric
+reporting to a third party, namely, Heroku's Application Metrics endpoint
+available over HTTPS. It's capable of receiving metrics via statsd, and
+offers a simple Prometheus scraper as well. This document discusses the
+way in which all of these things interact.
+
+## Metrics
+
+Those familiar with standard operational metric terminology won't be
+surprised to learn that agentmon has a concept of counters and gauges.
+These behave very similarly to counters and gauges from
+the [statsd][statsd] "protocol". 
+
+That is to say, for a time period defined by the flush interval, the
+value of a named counter metric is the sum of all values reported for
+that name.
+
+For a gauge, in that same time period, the last value received for a
+given name is the value that will be reported.
+
+Outside of the statsd "protocol," agentmon is also capable of handling
+monotonically increasing counters as well. Internally, these are called
+"derived counters," and they are simply flushed as regular counters. 
+Each time a measurement is observed, the value added to the counter is
+equal to `{observed value} - {previously observed value}`.
+
+In the [Etsy statsd][etsy-statsd] implementation, counters, and timers
+can have an attached sample rate that is typically used to reduce
+observations sent to statsd, accepting, possibly, a bit less accuracy
```

Comment:
` sent to statsd accepting, possibly, a bit less accuracy` three commas are too many IMHO